nir/src/main/scala/scala/scalanative/nir/serialization/BinaryDeserializer.scala

```diff
@@ -265,7 +268,7 @@ final class BinaryDeserializer(buffer: ByteBuffer) {
     case T.DoubleVal      => Val.Double(getDouble)
     case T.StructValueVal => Val.StructValue(getVals)
     case T.ArrayValueVal  => Val.ArrayValue(getType, getVals)
-    case T.CharsVal       => Val.Chars(getString)
+    case T.CharsVal       => Val.Chars(getBytes())
```

Comment:
Here we'll need a deserialization hack to be able to read previous versions or the NIR. The idea is that, if we read an .nir file produced by a previous version, we'll do `getString`, and then process the escapes of that string like the previous linker used to do, to produce bytes. And otherwise we just call `getBytes()`.

How do we know whether we're reading a previous version? It's stored in the `Prelude`. Inside `val header = ...`, we call `Prelude.readFrom(), which returns a value of type `Prelude` that contains this information. Currently we throw it away because we don't need it, but now we'll have to store it.

Reply:
Hey, 
I'm not really sure is it needed in this place, previously this new `getBytes` method is simply extracted logic from previous version of `getString`. Currently getString is defined as 
```  
private def getString(): String = {
    new String(getBytes(), "UTF-8")
  }
```

altough as I understand inside `CodeGen` we should change current logic to 
```
...
      case v :Val.Chars =>
        if(v.revision < 7) legacyGenChars(v.stringValue) //here we're using previous logic
        else genChars(v.bytes) // we're using new logic which assumes string is escaped
```
Am i correct?

Reply:
We cannot have version-dependent behavior in CodeGen. The problem is that things can get inlined from one place to another, so a `Val.Chars` may be copied over to a `Def` that was originally with a different `revision`.

The last and only moment we can use version-dependent logic is in the deserializer. After that, all the NIR must be assumed to be in the latest version, and be handled as such.